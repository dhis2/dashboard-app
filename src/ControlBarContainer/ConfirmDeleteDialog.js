import React from 'react';
import PropTypes from 'prop-types';
import { Trans } from 'react-i18next';
import Dialog from 'material-ui/Dialog';
import FlatButton from '../widgets/FlatButton';
import PrimaryButton from '../widgets/PrimaryButton';

const ConfirmDeleteDialog = (
    { dashboardName, onDeleteConfirmed, onContinueEditing, open },
    context
) => {
    const actions = [
        <FlatButton onClick={onDeleteConfirmed}>
            {context.i18n.t('Delete')}
        </FlatButton>,
        <PrimaryButton onClick={onContinueEditing}>
            {context.i18n.t('Continue editing')}
        </PrimaryButton>,
    ];

    return (
        <Dialog
            title={context.i18n.t('Confirm delete dashboard')}
            actions={actions}
            modal={true}
            open={open}
        >
            <Trans>
                Are you sure you want to delete dashboard{' '}
                <strong>{{ dashboardName }}</strong>?
            </Trans>
        </Dialog>
    );
};

ConfirmDeleteDialog.contextTypes = {
    i18n: PropTypes.object,
};

export default ConfirmDeleteDialog;
