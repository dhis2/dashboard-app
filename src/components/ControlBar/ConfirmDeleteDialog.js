import React from 'react';
import i18n from '@dhis2/d2-i18n';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import { Button } from '@dhis2/ui-core';

export const ConfirmDeleteDialog = ({
    dashboardName,
    onDeleteConfirmed,
    onContinueEditing,
    open,
}) => {
    const actions = [
        <Button key="cancel" secondary onClick={onContinueEditing}>
            {i18n.t('Cancel')}
        </Button>,
        <Button key="delete" destructive onClick={onDeleteConfirmed}>
            {i18n.t('Delete')}
        </Button>,
    ];

    return (
        <Dialog open={open} onClose={onContinueEditing}>
            <DialogTitle style={{ fontSize: '1.25rem', fontWeight: '450' }}>
                {i18n.t('Delete dashboard')}
            </DialogTitle>
            <DialogContent>
                <DialogContentText style={{ fontSize: '0.875rem' }}>
                    {i18n.t(
                        'Deleting dashboard "{{ dashboardName }}" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?',
                        { dashboardName }
                    )}
                </DialogContentText>
            </DialogContent>
            <DialogActions>{actions}</DialogActions>
        </Dialog>
    );
};

export default ConfirmDeleteDialog;
