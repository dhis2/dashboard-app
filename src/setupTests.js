import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const userSettings = {
    keyDbLocale: 'ponyLang',
};

export function getStubContext() {
    return {
        i18n: {
            t: () => {},
        },
        d2: {
            currentUser: {
                firstName: 'Mark the Ghost',
                surname: 'Polak',
                userSettings: {
                    get: key => userSettings[key],
                },
            },
        },
    };
}
