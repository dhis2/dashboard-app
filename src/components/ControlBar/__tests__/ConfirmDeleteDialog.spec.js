import React from 'react';
import { shallow } from 'enzyme';
import DialogActions from '@material-ui/core/DialogActions';
import { ConfirmDeleteDialog } from '../ConfirmDeleteDialog';
import { getStubContext } from '../../../setupTests';

describe('ConfirmDeleteDialog', () => {
    let props;
    let shallowDialog;
    const dialog = () => {
        if (!shallowDialog) {
            shallowDialog = shallow(<ConfirmDeleteDialog {...props} />, {
                context: getStubContext(),
            });
        }
        return shallowDialog;
    };

    beforeEach(() => {
        props = {
            dashboardName: '',
            onDeleteConfirmed: jest.fn(),
            onContinueEditing: jest.fn(),
            open: false,
        };
        shallowDialog = undefined;
    });

    it('matches the snapshot', () => {
        expect(dialog()).toMatchSnapshot();
    });

    it('renders a Button with action onContinueEditing', () => {
        expect.assertions(1);
        dialog()
            .find(DialogActions)
            .children()
            .forEach(actionEl => {
                if (actionEl.key() === 'cancel') {
                    expect(actionEl.prop('onClick')).toBe(
                        props.onContinueEditing
                    );
                }
            });
    });

    it('renders a Button with action onDeleteConfirmed', () => {
        expect.assertions(1);
        dialog()
            .find(DialogActions)
            .children()
            .forEach(actionEl => {
                if (actionEl.key() === 'delete') {
                    expect(actionEl.prop('onClick')).toBe(
                        props.onDeleteConfirmed
                    );
                }
            });
    });
});
