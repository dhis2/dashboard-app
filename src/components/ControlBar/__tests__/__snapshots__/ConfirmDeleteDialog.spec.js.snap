// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDeleteDialog matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmDeleteDialog
    dashboardName=""
    onContinueEditing={[MockFunction]}
    onDeleteConfirmed={[MockFunction]}
    open={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(DialogTitle)
          style={
            Object {
              "fontSize": "1.25rem",
              "fontWeight": "450",
            }
          }
        >
          Delete dashboard
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)
            style={
              Object {
                "fontSize": "0.875rem",
              }
            }
          >
            Deleting dashboard "" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?
          </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <Button
            onClick={[MockFunction]}
            secondary={true}
            type="button"
          >
            Cancel
          </Button>
          <Button
            destructive={true}
            onClick={[MockFunction]}
            type="button"
          >
            Delete
          </Button>
        </WithStyles(DialogActions)>,
      ],
      "onClose": [MockFunction],
      "open": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Delete dashboard",
          "style": Object {
            "fontSize": "1.25rem",
            "fontWeight": "450",
          },
        },
        "ref": null,
        "rendered": "Delete dashboard",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(DialogContentText)
            style={
              Object {
                "fontSize": "0.875rem",
              }
            }
          >
            Deleting dashboard "" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?
          </WithStyles(DialogContentText)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Deleting dashboard \\"\\" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?",
            "style": Object {
              "fontSize": "0.875rem",
            },
          },
          "ref": null,
          "rendered": "Deleting dashboard \\"\\" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              onClick={[MockFunction]}
              secondary={true}
              type="button"
            >
              Cancel
            </Button>,
            <Button
              destructive={true}
              onClick={[MockFunction]}
              type="button"
            >
              Delete
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "cancel",
            "nodeType": "function",
            "props": Object {
              "children": "Cancel",
              "onClick": [MockFunction],
              "secondary": true,
              "type": "button",
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "delete",
            "nodeType": "function",
            "props": Object {
              "children": "Delete",
              "destructive": true,
              "onClick": [MockFunction],
              "type": "button",
            },
            "ref": null,
            "rendered": "Delete",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(DialogTitle)
            style={
              Object {
                "fontSize": "1.25rem",
                "fontWeight": "450",
              }
            }
          >
            Delete dashboard
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
              style={
                Object {
                  "fontSize": "0.875rem",
                }
              }
            >
              Deleting dashboard "" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <Button
              onClick={[MockFunction]}
              secondary={true}
              type="button"
            >
              Cancel
            </Button>
            <Button
              destructive={true}
              onClick={[MockFunction]}
              type="button"
            >
              Delete
            </Button>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [MockFunction],
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Delete dashboard",
            "style": Object {
              "fontSize": "1.25rem",
              "fontWeight": "450",
            },
          },
          "ref": null,
          "rendered": "Delete dashboard",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(DialogContentText)
              style={
                Object {
                  "fontSize": "0.875rem",
                }
              }
            >
              Deleting dashboard "" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?
            </WithStyles(DialogContentText)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Deleting dashboard \\"\\" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?",
              "style": Object {
                "fontSize": "0.875rem",
              },
            },
            "ref": null,
            "rendered": "Deleting dashboard \\"\\" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                onClick={[MockFunction]}
                secondary={true}
                type="button"
              >
                Cancel
              </Button>,
              <Button
                destructive={true}
                onClick={[MockFunction]}
                type="button"
              >
                Delete
              </Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "cancel",
              "nodeType": "function",
              "props": Object {
                "children": "Cancel",
                "onClick": [MockFunction],
                "secondary": true,
                "type": "button",
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "delete",
              "nodeType": "function",
              "props": Object {
                "children": "Delete",
                "destructive": true,
                "onClick": [MockFunction],
                "type": "button",
              },
              "ref": null,
              "rendered": "Delete",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "d2": Object {
        "currentUser": Object {
          "firstName": "Mark the Ghost",
          "surname": "Polak",
          "userSettings": Object {
            "get": [Function],
          },
        },
      },
      "i18n": Object {
        "t": [Function],
      },
    },
  },
}
`;
